sdk/servicebus/service-bus/src/track2/serviceBusReceiverClient.ts

```diff
@@ -0,0 +1,526 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ServiceBusMessage, ReceiveMode } from "../serviceBusMessage";
+import { ClientEntityContext } from "../clientEntityContext";
+import { generate_uuid } from "rhea-promise";
+import { ClientType } from "../client";
+import { InternalSessionReceiver, InternalReceiver } from "../internalReceivers";
+import {
+  MessageHandlers,
+  MessageIterator,
+  Session,
+  ContextType,
+  ContextWithSettlement,
+  QueueAuth,
+  SubscriptionAuth,
+  isSession,
+  MessageAndContext,
+  Message,
+  ReceiveBatchOptions,
+  IterateMessagesOptions,
+  StreamMessagesOptions
+} from "./models";
+import { createConnectionContext, convertToInternalReceiveMode } from "./constructorHelpers";
+import { RuleDescription, CorrelationFilter } from "../core/managementClient";
+import { ServiceBusClientOptions } from "../old/serviceBusClient";
+import { ConnectionContext } from "../connectionContext";
+
+/**
+ *A receiver client that handles sessions, including renewing the session lock.
+ */
+// TODO: could extend NonSessionReceiverClient...?
+export interface SessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
+  /**
+   * Streams messages to message handlers.
+   * @param handler A handler that gets called for messages and errors.
+   * @param options Options for streamMessages.
+   */
+  streamMessages(
+    handlers: MessageHandlers<ContextType<LockModeT>>,
+    options?: StreamMessagesOptions
+  ): void;
+  /**
+   * Returns an iterator that can be used to receive messages from Service Bus.
+   * @param options Options for iterateMessages.
+   */
+  iterateMessages(options?: IterateMessagesOptions): MessageIterator<ContextType<LockModeT>>;
+
+  /**
+   * Receives, at most, `maxMessages` worth of messages.
+   * @param maxMessages The maximum number of messages to accept.
+   * @param maxWaitTimeInSeconds The maximum time to wait, in seconds, for messages to arrive.
+   * @param options Options for receiveBatch.
+   */
+  receiveBatch(
+    maxMessages: number,
+    maxWaitTimeInSeconds?: number,
+    options?: ReceiveBatchOptions
+  ): Promise<Message[]>;
+  /**
+   * Renews the lock on the session.
+   */
+  renewSessionLock(): Promise<Date>;
+  /**
+   * Closes the client.
+   */
+  close(): Promise<void>;
+  /**
+   * Methods related to service bus diagnostics.
+   */
+  diagnostics: {
+    /**
+     * Peek within a queue or subscription.
+     * @param maxMessageCount The maximum number of messages to retrieve.
+     */
+    peek(maxMessageCount?: number): Promise<Message[]>;
```

Comment:
Did we determine for sure whether peek prevents any other receivers from receiving from the same session?

Reply:
I'll double-check.

Reply:
Double-checked and it's still true. I filed an issue though because our error message is bad when it does happen:

In .NET: 
   Unhandled exception. Microsoft.Azure.ServiceBus.SessionCannotBeLockedException: The requested session 'helloworld' cannot be accepted. It may be locked by another receiver.

In ours:
    (node:24832) UnhandledPromiseRejectionWarning: MessagingError: Received sessionId 'undefined' does not match the provided sessionId 'helloworld' while creating the receiver 

Not nearly as descriptive.

Reply:
Filed #7618 to fix the message.